StudyHub/StudyHub/Controllers/GoogleAuthController.cs

```diff
@@ -14,10 +14,17 @@ public GoogleAuthController(IGoogleAuthService googleAuthService)
         _googleAuthService = googleAuthService;
     }
 
+    [HttpPost("sign-up")]
+    public async Task<IActionResult> GoogleSignUp([FromHeader(Name = "Authorization-Code")] string authorizationCode)
+    {
+        var result = await _googleAuthService.GoogleRegisterAsync(authorizationCode);
+        return Ok(result);
+    }
+
     [HttpPost("sign-in")]
     public async Task<IActionResult> GoogleSignIn([FromHeader(Name = "Authorization-Code")] string authorizationCode)
     {
-        var result = await _googleAuthService.GoogleLogin(authorizationCode);
+        var result = await _googleAuthService.GoogleLoginAsync(authorizationCode);
         return Ok(result);
```

Comment:
consistent naming for service and controller methods. Better to have Register and Login, I guess